pkg/backupdriver/backup_driver_controller.go

```diff
@@ -235,21 +235,77 @@ func (ctrl *backupDriverController) cloneFromSnapshot(cloneFromSnapshot *backupd
 }
 
 // Update the snapshot status phase
-func (ctrl *backupDriverController) updateSnapshotStatusPhase(snapshot *backupdriverapi.Snapshot, newPhase backupdriverapi.SnapshotPhase) error {
-	ctrl.logger.Infof("Entering updateSnapshotStatusPhase: %s/%s, Phase %s", snapshot.Namespace, snapshot.Name, newPhase)
+func (ctrl *backupDriverController) updateSnapshotStatusPhase(ctx context.Context, snapshotNs string, snapshotName string,
+	newPhase backupdriverapi.SnapshotPhase, snapshotStatusFields map[string]interface{}) (*backupdriverapi.Snapshot, error) {
+	ctrl.logger.Infof("Entering updateSnapshotStatusPhase: %s/%s, Phase %s", snapshotNs, snapshotName, newPhase)
+
+	// Retrieve the latest version of Snapshot and update the status.
+	snapshot, err := ctrl.backupdriverClient.Snapshots(snapshotNs).Get(ctx, snapshotName, metav1.GetOptions{})
+	if err != nil {
+		ctrl.logger.Errorf("updateSnapshotStatusPhase: Failed to retrieve the latest snapshot state, the snapshot operation was however successful %v", err)
```

Comment:
How can you be sure that "the snapshot operation was however successful"?  "updateSnapshotStatusPhase" is a utility function that can be called to update snapshot status after snapshot operation fails as well.  This line should be removed.

Reply:
You are right, considering updateSnapshotStatusPhase can be called from different methods the log is misleading. I'll remove it.